lib/schedule.js

```diff
@@ -266,11 +266,42 @@ function RecurrenceRule(year, month, date, dayOfWeek, hour, minute, second) {
 }
 
 RecurrenceRule.prototype.validate = function() {
-  // TODO: validation
+  if (!recurValidateProperty(this.year, 0, Infinity)) {
+    return false;
+  }
+
+  if (!recurValidateProperty(this.month, 0, 11)) {
+    return false;
+  }
+
+  if (!recurValidateProperty(this.dayOfWeek, 0, 6)) {
+    return false;
+  }
+
+  if (!recurValidateProperty(this.date, 0, 30)) {
+    return false;
+  }
+
+  if (!recurValidateProperty(this.hour, 0, 23)) {
+    return false;
+  }
+
+  if (!recurValidateProperty(this.minute, 0, 59)) {
+    return false;
+  }
+
+  if (!recurValidateProperty(this.second, 0, 59)) {
+    return false;
+  }
+
   return true;
 };
 
 RecurrenceRule.prototype.nextInvocationDate = function(base) {
+  if (!this.validate()) {
+    throw new Error('Invalid RecurrenceRule');
+  }
```

Comment:
It would we great that validate also returned info about what field is invalid. Then we could throw and error with a message like this: `Invalid RecurrenceRule. Field year is out of range`